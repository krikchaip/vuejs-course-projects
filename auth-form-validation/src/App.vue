<template>
  <div>
    <Navigation>
      <template v-if="$store.getters.isAuthenticated">
        <router-link to="/dashboard">Dashboard</router-link>
        <button @click="$store.dispatch('logout-user')">Logout</button>
      </template>
      <template v-else>
        <router-link to="/signup">Sign Up</router-link>
        <router-link to="/signin">Sign In</router-link>
      </template>
    </Navigation>
    <div class="view-container">
      <router-view/>
    </div>
  </div>
</template>

<script>
  import Navigation from '@/components/Navigation'

  export default {
    components: { Navigation }
  }
</script>

<style>
  @import '~@/assets/styles.css';
</style>
