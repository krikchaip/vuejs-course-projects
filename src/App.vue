<template>
  <div class="container">
    <the-header class="mt-5 mb-3"/>
    <quote-add/>
    <div class="row no-gutters">
      <quote>Test</quote>
    </div>
    <div class="alert alert-info text-center">
      <small>Info: Click on a Quote to delete it.</small>
    </div>
  </div>
</template>

<script>
  export default {
    components: {
      TheHeader: require('@/components/TheHeader').default,
      Quote: require('@/components/Quote').default,
      QuoteAdd: require('@/components/QuoteAdd').default
    }
  }
</script>
