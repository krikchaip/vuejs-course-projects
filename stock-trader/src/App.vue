<template>
  <div class="container">
    <Navigation class="my-3">
      <div class="d-flex mr-3">
        <ActionEndDay/>
        <ActionSaveLoad/>
      </div>
    </Navigation>
    <transition name="slide-fade" mode="out-in">
      <router-view/>
    </transition>
  </div>
</template>

<script>
  import Navigation from '@/components/Navigation'
  import ActionEndDay from '@/components/ActionEndDay'
  import ActionSaveLoad from '@/components/ActionSaveLoad'

  export default {
    components: { Navigation, ActionEndDay, ActionSaveLoad }
  }
</script>

<style>
  @import '~@/assets/bootstrap.min.css';

  * {
    user-select: none;
  }

  .slide-fade-enter-active {
    transition: all 0.2s ease-out;
  }

  .slide-fade-leave-active {
    transition: all 0.2s ease-out;
  }

  .slide-fade-enter, .slide-fade-leave-to {
    transform: translateY(10px);
    opacity: 0;
  }
</style>
