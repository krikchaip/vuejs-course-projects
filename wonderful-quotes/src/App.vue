<template>
  <div class="container">
    <the-header
      class="mt-5 mb-3"
      :current="quotes.length"
      :max="MAX_QUOTES"
    />
    <quote-add @new-quote="append_new"/>
    <div class="row no-gutters">
      <quote
        v-for="(quote_text, idx) in quotes"
        :key="idx"
        @click.native="remove_self(idx)">
        {{ quote_text }}
      </quote>
    </div>
    <div class="alert alert-info text-center">
      <small>Info: Click on a Quote to delete it.</small>
    </div>
  </div>
</template>

<script>
  export default {
    components: {
      TheHeader: require('@/components/TheHeader').default,
      Quote: require('@/components/Quote').default,
      QuoteAdd: require('@/components/QuoteAdd').default
    },
    data: () => ({
      MAX_QUOTES: 10,
      quotes: ['Just a Quote to start with something!']
      // quotes: [
      //   'Just',
      //   'a',
      //   'Quote',
      //   'to',
      //   'start',
      //   'with',
      //   'something!',
      //   'winner',
      //   'za',
      //   'kub'
      // ]
    }),
    methods: {
      append_new(quote) {
        this.quotes.length < this.MAX_QUOTES
          ? this.quotes.push(quote)
          : alert('Too many Quotes! Delete some before adding new ones!')
      },
      remove_self(idx) {
        this.quotes.splice(idx, 1)
      }
    }
  }
</script>
