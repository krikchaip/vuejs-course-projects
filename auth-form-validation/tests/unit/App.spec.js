import { shallowMount, RouterLinkStub, Wrapper } from '@vue/test-utils'
// @ts-ignore
import App from '@/App'

describe('Nav links state', () => {
  /**
   * @param {Wrapper} wrapper
   * @return {Array<*>}
   */
  function findAllLinksProps(wrapper) {
    return wrapper.findAll(RouterLinkStub).wrappers.map(w => w.props())
  }

  /** @type {Wrapper} */
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(App, {
      stubs: {
        'router-link': RouterLinkStub,
        'router-view': true
      },
      methods: { _created() {} },
      mocks: {
        $store: {
          getters: {
            isAuthenticated: null
          }
        }
      }
    })
  })

  it('show sign-up, sign-in button when not logged in', () => {
    wrapper.vm.$store.getters.isAuthenticated = false
    wrapper.vm.$forceUpdate()

    expect(findAllLinksProps(wrapper).map(w => w.to).sort())
      .toMatchObject(['/signup', '/signin'].sort())
  })

  it('show dashboard, logout button when logged in', () => {
    wrapper.vm.$store.getters.isAuthenticated = true
    wrapper.vm.$forceUpdate()

    expect(wrapper.find({ name: 'Navigation' }).html()).toMatchSnapshot()
  })
})
