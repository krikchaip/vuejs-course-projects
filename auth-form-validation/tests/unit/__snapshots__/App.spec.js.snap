// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav links state show dashboard, logout button when logged in 1`] = `
<navigation-stub>
  <a>Dashboard</a>
  <button>Logout</button>
</navigation-stub>
`;
